### PR TITLE
fix: use deferred value to avoid suspense issues with inputting text

### PR DESCRIPTION
**Pull request by natew**
use deferred value to avoid suspense issues with inputting text during supsended render causing errors

## Description

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2525/[web][ipados][search]-deleting-any-search-causes-an-error-in-the-page
_Slack thread:_
_Relevant docs:_

### Before

If you hit backspace on a search it breaks search pretty often

### After

No break

### Reproducing the error

1. 

### QA (ie manual testing)

- [ ] N/A

### Automated testing

- [ ] Unit test
- [ ] Integration/E2E test

### CODE REVIEW

File: src/components/Suspense/SuspenseWithPreviousRenderAsFallback.tsx [hunk 1]
```diff
@@ -10,5 +10,7 @@ import React, { Suspense } from 'react'
 
 export const SuspenseWithPreviousRenderAsFallback = (props: { children: React.ReactNode }) => {
   const previousContents = usePrevious(props.children)
-  return <Suspense fallback={previousContents ?? null}>{props.children}</Suspense>
+  const previousDeferred = useDeferredValue(previousContents)
```
Comment (natew) [inline src/components/Suspense/SuspenseWithPreviousRenderAsFallback.tsx:13]: evidently this works? https://stackoverflow.com/a/76173194

Comment (JackShort) [reply src/components/Suspense/SuspenseWithPreviousRenderAsFallback.tsx:13]: do we need `usePrevious` and `useDeferredValue` or would this work with just `useDeferredValue`?

Comment (cbachmeier) [reply src/components/Suspense/SuspenseWithPreviousRenderAsFallback.tsx:13]: Confirmed that just using `useDeferredValue` gives the desired search behaviour without crashing on iPad

Comment (cbachmeier) [reply src/components/Suspense/SuspenseWithPreviousRenderAsFallback.tsx:13]:
```suggestion
  const previousDeferred = useDeferredValue(props.children)
```

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: src/components/Suspense/SuspenseWithPreviousRenderAsFallback.tsx (hunk 1 above)
Comment (natew) [reply src/components/Suspense/SuspenseWithPreviousRenderAsFallback.tsx:13]: we may need previous, this area also fixes another issue we had in search, let me make sure tests pass and it works still without

Comment (natew) [reply src/components/Suspense/SuspenseWithPreviousRenderAsFallback.tsx]: confirm it works this way!

**Comment by natew** [comment]
added flake fixes as requested by @zzmp it's not beautiful but from what i can see in cypress its just cypress being flaky for whatever reason

File: cypress/e2e/universal-search.test.ts [hunk 2]
```diff
@@ -25,31 +25,42 @@ describe('Universal search bar', () => {
     cy.location('hash').should('equal', '#/tokens/ethereum/0x1f9840a85d5af5bf1d1762f925bdaddc4201f984')
   })
 
-  it('should go to the selected result when recent results are shown', () => {
-    // Search for uni token by name.
-    getSearchBar().type('uni')
-    cy.get('[data-cy="searchbar-token-row-UNI"]')
+  it(
```
Comment (cbachmeier) [inline cypress/e2e/universal-search.test.ts:28]: these changes should be in their own PR and not coupled to this hotfix

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: cypress/e2e/universal-search.test.ts (hunk 2 above)
Comment (natew) [reply cypress/e2e/universal-search.test.ts:28]: it's failing this test though, this tests the same element we're fixing

Comment (natew) [reply cypress/e2e/universal-search.test.ts:28]: more context: i added this test when i implemented the SuspenseFallback stuff, basically this test was made for the same code were fixing here

Comment (cbachmeier) [reply cypress/e2e/universal-search.test.ts:28]: These are two different fixes.
1. Fixes the suspense issue causing a page crash
2. Fixes a flaky test relating to the searchbar that is currently failing on main
These should be their own PRs. The PR that fixes the flaky test by adding retries should not have the rider clause of altering use Suspense behaviour or vice versa because they touch the same component. They're two separate changes.

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: cypress/e2e/universal-search.test.ts (hunk 2 above)
Comment (natew) [reply cypress/e2e/universal-search.test.ts:28]: The flaky test was actually added to cover the thing that's erroring. I'll split it out, but they are one and the same (test is testing the area that breaks, was added to cover the thing that broke). I do think adding yet another test to test backspace would be good here as well.

**Review by cbachmeier — APPROVED** [review]

**Review by zzmp — APPROVED** [review]
I was not able to reproduce the error, but this looks good nonetheless.
